KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java

```diff
@@ -143,11 +143,12 @@ public void run() {
                         setGroupPermision(filePath);
                     }
 
-                     updatedEntry = appendRecording(liveEntry, liveAsset.id, index, filePath, appendTime, lastChunkFlag);
-                    if (updatedEntry != null){
-                        liveEntry = updatedEntry;
+                    if (!KalturaAPI.getKalturaAPI().isNewRecordingEnabled(liveEntry)) {
```

Comment:
it's bad practive todo API call over and over when you already know the answer...